modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItComputeSystemViewTest.java

```diff
@@ -0,0 +1,309 @@
+        Ignite targetNode = CLUSTER.node(0);
+
+        try {
+            ClockService clockService = unwrapIgniteImpl(targetNode).clockService();
+
+            long tsBefore = clockService.now().getPhysical();
+
+            TaskExecution<Void> execution = entryNode.compute()
+                    .submitMapReduce(TaskDescriptor.builder(MapReduceTaskCustom.class).build(), null);
+
+            await().until(execution::stateAsync, willBe(taskStateWithStatus(TaskStatus.EXECUTING)));
+
+            long tsAfter = clockService.now().getPhysical();
+
+            String query = "SELECT * FROM SYSTEM.COMPUTE_TASKS WHERE STATUS = ?";
+
+            List<List<Object>> res = sql(0, query, EXECUTING.name());
+
+            assertThat(res.size(), is(2));
+            List<String> execNodes = List.of(CLUSTER.node(0).name(), CLUSTER.node(1).name());
+
+            verifyQueryInfo(res.get(0), execNodes, EXECUTING.name(), tsBefore, tsAfter);
+            verifyQueryInfo(res.get(1), execNodes, EXECUTING.name(), tsBefore, tsAfter);
+
+            IgniteTestUtils.await(execution.cancelAsync());
+        } finally {
+            closeQuiet(entryNode);
+        }
+    }
+
+    private static ClusterNode clusterNode(Ignite node) {
+        return unwrapIgniteImpl(node).node();
+    }
+
+    private static class InfiniteMapReduceJob implements ComputeJob<Void, Void> {
+        @Override
+        public CompletableFuture<Void> executeAsync(JobExecutionContext context, Void input) {
+            return new CompletableFuture<>();
+        }
+    }
+
+    private static class MapReduceTaskCustom implements MapReduceTask<Void, Void, Void, Void> {
+        @Override
+        public CompletableFuture<List<MapReduceJob<Void, Void>>> splitAsync(TaskExecutionContext taskContext, @Nullable Void input) {
+            return completedFuture(List.of(
+                    MapReduceJob.<Void, Void>builder()
+                            .jobDescriptor(JobDescriptor.builder(InfiniteMapReduceJob.class).build())
+                            .nodes(taskContext.ignite().clusterNodes())
+                            .build()
+            ));
+        }
+
+        @Override
+        public CompletableFuture<Void> reduceAsync(TaskExecutionContext taskContext, Map<UUID, Void> results) {
+            return completedFuture(null);
+        }
+    }
+
+    private static class InfiniteJob implements ComputeJob<Void, Void> {
+        @Override
+        public @Nullable CompletableFuture<Void> executeAsync(JobExecutionContext context, @Nullable Void arg) {
+            while (true) {
+                try {
+                    Thread.sleep(500);
+                } catch (InterruptedException e) {
+                    // No op, just return from loop
+                    break;
+                }
+            }
+
+            return null;
+        }
+    }
+
+    private static void verifyQueryInfo(
```

Comment:
may be it's better to rename this method to something like
`verifyComputeJobState` or `verifyExecutionState` or `verifyViewRowData`.

The method with this name in `ItSqlQueriesSystemViewTest` checks "query info" (containing in `QueryInfo` class).

Reply:
i have nothing against current naming, it`s correct as for me, but ok i change it